packages/astro/src/@types/astro.ts

```diff
@@ -1640,6 +1640,25 @@ export interface AstroUserConfig {
 		 */
 		contentCollectionCache?: boolean;
 
+		/**
+		 * @docs
+		 * @name experimental.contentCollectionJSONSchema
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.5.0
+		 * @description
+		 * Enables generation of JSON Schema files for content collections of type `data`.
+		 *
+		 * ```js
+		 * {
+		 * 	experimental: {
+		 * 		contentCollectionJSONSchema: true,
+		 * 	},
+		 * }
+		 * ```
+		 */
+		contentCollectionJSONSchema?: boolean;
```

Comment:
How do we feel about this experimental flag name?

Reply:
Looks good. Nit for me: `contentCollectionJsonSchema`